modules/data.land/R/match_species_id.R

```diff
@@ -0,0 +1,65 @@
+#' Match BETY species ID.
+#'
+#' Parses species codes in input data and matches them with the BETY species ID.
+#' 
+#' @param input_codes Character vector of species codes
+#' @param format_name Species code format name (character, length 1)
+#' @param betydb \code{dplyr} \code{src} object containing BETY connection 
+#' @param custom_translation_path File path to translation
+#' @return \code{data.frame} containing the following columns:
```

Comment:
That's an option, but in that case format should be an explicit argument, since it's required but not documented. Personally, I don't like that option because it means you need to make a Format entry for both the data file and the species translation file, or you have to make a Format list object, which is also complicated. I think this is a case where it's fine to require a standard file format, since (unlike the veg data itself) no one is going to have this table just sitting around already (since the key column is the bety species id's).  All of this is making me thing that passing the data frame itself, rather than the path, will probably be the more common usage.

Once this function is in place, it would be very useful to think about the utilities we'll need to provide to make it easy for people to make these tables -- this seems like a great use case for the ROpenSci traits package for pulling those id's out of BETYdb (plus taxize, etc).